### PR TITLE
Add GraphQL chapter

### PR DESCRIPTION
**Pull request by gabriel-glo**

### CODE REVIEW

File: WordPress as API/WordPress GraphQL API.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+WordPress REST API can be extended to support [GraphQL](https://graphql.org/). GraphQL is a query language for your API and it acts as a separate layer between WordPress REST API and application that is displaying data from WordPress.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:1]:
```suggestion
WordPress REST API can be extended to support [GraphQL](https://graphql.org/). GraphQL is a query language for your API and it acts as a separate layer between the WordPress REST API and the application using data from the API.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+With GraphQL we are enabling apps to define their own queries and what data will be retrieved from the API. GraphQL queries allow getting data from multiple resources at once, which is a big plus over the standard REST API way. With REST API you'd either have multiple routes and send requests to each one individually, or have a monolithic route that will already contain all the resources that you may need.
+
+### GraphQL in WordPress
+
+GraphQL in WordPress can be enables via [WPGraphQL plugin](https://www.wpgraphql.com/). It is a free plugin, hosted on the WordPress plugin repository. It comes with a pretty useful GraphiQL tool that allows you to preview GraphQL schema and test out some queries.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:7]:
```suggestion
GraphQL in WordPress can be enabled via the [WPGraphQL plugin](https://www.wpgraphql.com/). It is a free plugin, hosted on the WordPress plugin repository. It comes with a pretty useful _GraphiQL_ tool that allows you to preview GraphQL schemas and test queries.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+![GraphiQL initial view](../img/graphiql-1.png "GraphiQL initial view")
+
+To test the queries click on resources on the left sidebar to drill down in their structure and see what is available. For example, to get all posts and show their title, link, excerpt and content, the query would look like this:
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:11]:
```suggestion
To test the queries click on resources in the left sidebar. You can drill down in their structure and see what is available. For example, to get all posts and show their title, link, excerpt and content, the query would look like this:
```

File: WordPress as API/WordPress GraphQL API.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      excerpt
+      content
+    }
+  }
+}
+```
+
+![GraphiQL query view](../img/graphiql-2.png "GraphiQL query view")
+
+To combine multiple resources in one query you just continue to select the in the sidebar or write them down in query editor. For this example we will query all post categories and all posts.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:28]:
```suggestion
To combine multiple resources in one query you can keep selecting resources the in the sidebar or add them in the query editor. For this example we will query all post categories and all posts.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+![GraphiQL query view](../img/graphiql-3.png "GraphiQL query view")
+
+This is just a start of interacting with GraphiQL and more examples and guides can be found in plugins [documentation](https://www.wpgraphql.com/docs/).
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:32]:
```suggestion
This is just a start of interacting with GraphiQL and more examples and guides can be found in the plugin's [documentation](https://www.wpgraphql.com/docs/).
```

File: WordPress as API/WordPress GraphQL API.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+Apps that will use WordPress GraphQL API will need to send HTTP POST requests to the GraphQL endpoint and provide the query in request body. WPGraphQL plugin registers this endpoint and by default it is `http://your-domain.com/graphql/`.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:34]:
```suggestion
Apps that use the WordPress GraphQL API will need to send HTTP POST requests to the GraphQL endpoint and provide the query in the request body. WPGraphQL plugin registers this endpoin, the default is `http://your-domain.com/graphql/`.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+To grasp some basic GraphQL concepts, read these chapters from the WPGraphQL documentation:
+* [Intro to GraphQL](https://www.wpgraphql.com/docs/intro-to-graphql/)
+* [WPGraphQL Concepts](https://www.wpgraphql.com/docs/wpgraphql-concepts/)
+
+### Adding new resources to the WordPress GraphQL
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:40]:
```suggestion
### Adding new resources to WordPress GraphQL
```

File: WordPress as API/WordPress GraphQL API.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+You can register new GraphQL types or add new fields to the existing GraphQL types.
+
+#### Adding new GraphQL post and taxonomy type
+
+To add a custom post or a taxonomy types to the GraphQL schema you need to provide additional arguments to the register functions.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:46]:
```suggestion
To add a custom post or a taxonomy type to the GraphQL schema you need to provide additional arguments to the register functions.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+	'show_in_graphql' => true,
+	'graphql_single_name' => 'singleName',
+	'graphql_plural_name' => 'pluralName',
+	// ... other post type or taxonomy arguments
+]
+```
+
+To add or remove post types or taxonomies that are created by default or by a third-party plugins, you will need to use [register_post_type_args](https://developer.wordpress.org/reference/hooks/register_post_type_args/) or [register_taxonomy_args](https://developer.wordpress.org/reference/hooks/register_taxonomy_args/) filter and return modified arguments array.
+
+GraphQL will use the single and plural name to register new GraphQL [types](https://graphql.org/learn/schema/#type-system) which will be used in queries.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:60]:
```suggestion
GraphQL will use the single and plural names to register new GraphQL [types](https://graphql.org/learn/schema/#type-system) which will be used in queries.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+#### Adding new GraphQL type
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:62]:
```suggestion
#### Adding a new GraphQL type
```

File: WordPress as API/WordPress GraphQL API.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+
+WPGraphQL comes with a lot of registered types and you can use them to register more complex fields. Most of the time you'll use [scalar](https://www.wpgraphql.com/docs/default-types-and-fields/#scalars) (simple) types but you can register your own custom types. More on this can be read in this [documentation page](https://www.wpgraphql.com/docs/default-types-and-fields).
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:64]:
```suggestion
WPGraphQL comes with a lot of registered types and you can use them to register more complex fields. Most of the time you'll use [scalar](https://www.wpgraphql.com/docs/default-types-and-fields/#scalars) (simple) types but you can register your own custom types. You can learn more about this in the [documentation](https://www.wpgraphql.com/docs/default-types-and-fields).
```

File: WordPress as API/WordPress GraphQL API.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+To register your own custom type for a field that needs to provide multiple values you will need to use [register_graphql_object_type](https://www.wpgraphql.com/functions/register_graphql_object_type/) inside the callback of [graphql_register_types](https://www.wpgraphql.com/actions/graphql_register_types/) action. You will need to provide a unique name for this new type and scalar fields that will be contained in this object type.
+
+
+#### Adding new GraphQL fields
+
+To add additional fields to the registered types you'd use [register_graphql_fields](https://www.wpgraphql.com/functions/register_graphql_fields/) or [register_graphql_field](https://www.wpgraphql.com/functions/register_graphql_field/) functions inside the callback of [graphql_register_types](https://www.wpgraphql.com/actions/graphql_register_types/) action.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:71]:
```suggestion
To add additional fields to the registered types you can use [register_graphql_fields](https://www.wpgraphql.com/functions/register_graphql_fields/) or [register_graphql_field](https://www.wpgraphql.com/functions/register_graphql_field/) functions inside the callback of [graphql_register_types](https://www.wpgraphql.com/actions/graphql_register_types/) action.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+				'resolve' => function($post, $args, $context, $info) { // Callback function that return the value of the field.
+					return \get_post_meta($post->ID, 'meta_field_key');
+				}
+			]
+		);
+	}
+}
+```
+
+The example above is for the field that returns a string value. You can register fields that will return more complex value. For example, you want to register image field that will return values of multiple images. For each image you will have URL of the image and the alt text.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:93]:
```suggestion
The example above is for the field that returns a string value. You can register fields that will return more complex values. For example, you might want to register an image field that will return values of multiple images. For each image you will have URL of the image and the alt text.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+
+To do that you need to first register a custom image type (remember that name must be unique) that will have two sub-fields: URL and alt text. In `register_graphql_field` function you will define a type of field as a list of image fields.
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:95]:
```suggestion
To do that you need to first register a custom image type (remember that the name must be unique!) that will have two sub-fields: URL and alt text. In the `register_graphql_field` function you will define a field type of of image fields.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+			'customImage',
+			[
+				'fields' => [
+					'imageURL' => [
+						'type' => 'String',
+						'description' => esc_html__('URL of the image', 'your-textdomain'),
+					],
+					'imageAlt' => [
+						'type' => 'String',
+						'description' => esc_html__('alt value of the image', 'your-textdomain'),
```
Comment (goranalkovic-infinum) [inline WordPress as API/WordPress GraphQL API.md:112]:
```suggestion
						'description' => esc_html__('Image alt text', 'your-textdomain'),
```

**Review by goranalkovic-infinum — DISMISSED** [review]
Couple of typos and suggestions, otherwise nice job 🎉

File: WordPress as API/WordPress GraphQL API.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
+WordPress REST API can be extended to support [GraphQL](https://graphql.org/). GraphQL is a query language for your API and it acts as a separate layer between the WordPress REST API and the application using data from the API.
+
+With GraphQL we are enabling apps to define their own queries and what data will be retrieved from the API. GraphQL queries allow getting data from multiple resources at once, which is a big plus over the standard REST API way. With REST API you'd either have multiple routes and send requests to each one individually, or have a monolithic route that will already contain all the resources that you may need.
+
+### GraphQL in WordPress
+
+GraphQL in WordPress can be enables via [WPGraphQL plugin](https://www.wpgraphql.com/). It is a free plugin, hosted on the WordPress plugin repository. It comes with a pretty useful GraphiQL tool that allows you to preview GraphQL schema and test out some queries.
```
Comment (dingo-d) [inline WordPress as API/WordPress GraphQL API.md]:
```suggestion
GraphQL in WordPress can be enabled via [WPGraphQL plugin](https://www.wpgraphql.com/). It is a free plugin, hosted on the WordPress plugin repository. It comes with a pretty useful GraphiQL tool that allows you to preview GraphQL schema and test out some queries.
```

File: WordPress as API/WordPress GraphQL API.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+}
+```
+
+![GraphiQL query view](../img/graphiql-2.png "GraphiQL query view")
+
+To combine multiple resources in one query you can keep selecting resources the in the sidebar or add them in the query editor. For this example we will query all post categories and all posts.
+
+![GraphiQL query view](../img/graphiql-3.png "GraphiQL query view")
+
+This is just a start of interacting with GraphiQL and more examples and guides can be found in plugins [documentation](https://www.wpgraphql.com/docs/).
```
Comment (dingo-d) [inline WordPress as API/WordPress GraphQL API.md:32]:
```suggestion
This is just the start of interacting with GraphiQL. More examples and guides can be found in plugin's [documentation](https://www.wpgraphql.com/docs/).
```

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left a few comments

File: WordPress as API/WordPress GraphQL API.md (hunk 14 above)
Comment (gabriel-glo) [reply WordPress as API/WordPress GraphQL API.md:95]: Doesn't work in this context

**Review by dingo-d — APPROVED** [review]

**Review by IvanGrginovInf — APPROVED** [review]

**Review by iruzevic — APPROVED** [review]

**Pull request merged**